target/linux/ramips/image/mt7620.mk

```diff
@@ -878,6 +901,28 @@ define Device/tplink_archer-c20i
 endef
 TARGET_DEVICES += tplink_archer-c20i
 
+define Device/tplink_re200-v1
+  SOC := mt7620a
+  DEVICE_VENDOR := TP-Link
+  DEVICE_PACKAGES := kmod-mt76x0e
+  DEVICE_MODEL := RE200
+  DEVICE_VARIANT := v1
+  IMAGES+= factory.bin
+  IMAGE/sysupgrade.bin := mktplinkfw sysupgrade -e -O
+  IMAGE/factory.bin := mktplinkfw factory -e -O
+  IMAGE_SIZE := 7936k
+  KERNEL_ENTRY := 0x8000c310
+  KERNEL_LOADADDR := 0x80000000
```

Comment:
According to the vendor fw these values are as follows:
Kernel load address: 0x00000080
Kernel entry point: 0x10c30080
did i miss something or are these just copied from another device?

Reply:
During boot of original firmware, exactly those values are reported on Serial and they are the same as mktplinkfw -i reports on the OEM firmware. However, the endianness needs to be converted, this is achieved with the "-e" switch during firmware creation.

Reply:
ah ok, thanks for the hint. i am currently working on the RE210v1, which is quite similar. i need to calculate my values again then :+1:

Reply:
ah, sorry, you're right, it displays your values, not mine - at least the same for the generated firmware and the OEM firmware.
I suppose it would work, but why do I want to pass wrong values when I can pass the correct ones and have the tool swap endianness?
This is what U-Boot reports on serial:
```
## Booting image at bc020000 ...
   Uncompressing Kernel Image ... OK
No initrd
## Transferring control to Linux (at address 8000c310) ...
## Giving linux memsize in MB, 64

Starting kernel ...
```

Reply:
yeah, there is no point in that. but i have another idea that might shrink your code down a bit. in mktplinkfw.c you can find a flash_layout struct. you might want to add another entry there containing your values as my device is using the same kernel addresses. that way, you can shorten the mktplinkfw define and just have to adjust the TPLINK_FLASHLAYOUT variable as "8Mlzma" is clearly not correct for both our devices. if you dont wannt the deal with that, i can adjust it in my pull request later on which would only require retesting your device.

Reply:
Good point! I thought about that previously, but I didn't want to modify the tool for just one device.

Somebody is working ok the RE200v3 which might also use the same layout. 

Now it makes sense, so I'll adapt the tool later today.

Reply:
I added the new layout "8Mmtk", because this name is also used in mktplinkfw2.c for MediaTek chipsets. It works fine for me :)